src/store.js

```diff
@@ -0,0 +1,64 @@
+import Vue from 'vue';
+import Vuex from 'vuex';
+import api from '@/api';
+
+Vue.use(Vuex);
+const store = new Vuex.Store({
+  state: {
+    messages: [],
+    refreshInterval: 30,
+    isLoading: false,
+    intervalId: null
+  },
+  mutations: {
+    setRefreshInterval(state, interval) {
+      state.refreshInterval = interval;
+    },
+    setIntervalId(state, intervalId) {
+      state.intervalId = intervalId;
+    },
+    setLoading(state, loading) {
+      state.isLoading = loading;
+    },
+    setMessages(state, messages) {
+      state.messages = messages;
+    },
+    clearIntervalTimeOut(state) {
+      clearInterval(state.intervalId);
+    }
+  },
+  actions: {
+    getMessages(context) {
+      context.commit('setLoading', true);
+
+      api.getMessages().then(messages => {
+        setTimeout(() => {
+          context.commit('setLoading', false);
+        }, 500);
+        context.commit('setMessages', messages);
+      });
+    },
+    cancelMessage(context, messageId) {
+      return new Promise(resolve => {
```

Comment:
no need to define a new Promise, `api.cancelMessage(messageId)` is already a Promise

`return api.cancelMessage(messageId);`